File: 1.2158.py
from lib import *
def main():
    IO_inf.freopen("pro.in")
    n=scanf("d")
    if n==1:
        print(0)
        return
    phi=Array(n)
    phi[1]=1
    isp=Array(n,dft=True)
    isp[1]=False
    cnt=0
    p=Array(n)
    for i in range(2,n):
        if isp[i]:
            p[cnt]=i
            cnt=cnt+1
            phi[i]=i-1
        for j in range(0,cnt):
            if i*p[j]>=n :
                break
            isp[i*p[j]]=False
            if i%p[j]==0 :
                phi[i*p[j]]=phi[i]*p[j]
                break
            else :
                phi[i*p[j]]=phi[i]*(p[j]-1)
    res=0
    for i in range(1,n):
        res=res+phi[i]
    print(res*2+1)

main()

File: 1.2158.cpp
#include<cstdio>
#include<cstring>
#include<cmath>
#include<algorithm>
using namespace std;
const int maxn=40005;
int n,res;
bool isp[maxn];
int p[maxn],cnt,phi[maxn];
int main()
{
#ifdef local
	freopen("pro.in","r",stdin);
#endif
	scanf("%d",&n);
	if(n==1) { puts("0"); return 0; }
	phi[1]=1;
	memset(isp,true,sizeof(isp));
	isp[1]=false;
	for(int i=2;i<=n-1;i++)
	{
		if(isp[i]) { p[cnt++]=i; phi[i]=i-1; }
		for(int j=0;j<cnt&&i*p[j]<=n-1;j++)
		{
			isp[i*p[j]]=false;
			if(i%p[j]==0)
			{
				phi[i*p[j]]=phi[i]*p[j];
				break;
			}
			else phi[i*p[j]]=phi[i]*(p[j]-1);
		}
	}
	for(int i=1;i<=n-1;i++) res+=phi[i];
	printf("%d\n",res*2+1);
	return 0;
}
